### ISSUE TITLE
Cannot find module "d3kit" when importing in TypeScript

### ISSUE THREAD

**Issue by morphatic** [twitter/d3kit#17]
I've been working on creating a [new TypeScript `.d.ts` definition file for d3kit v3.1.1](https://github.com/morphatic/DefinitelyTyped/tree/types-2.0/d3kit). I'm trying to just create a basic example of using d3kit in a TypeScript project. I followed the following process:

```
$ mkdir d3kit_test
$ cd d3kit_test
$ npm init
$ npm i --save d3 @types/d3 d3kit
$ typings install d3kit=github:morphatic/DefinitelyTyped/d3kit/index.d.ts#6cf5f4af25f4654f74e000367200823376e3233f --global --save
```

Then in `index.ts` I've got something like the following (based on the [example code in your README](https://github.com/twitter/d3kit/blob/master/README.md#b-create-a-reusable-chart)):

```typescript
import { SvgChart, helper } from "d3kit";

class SimpleBarChart extends SvgChart {
    static getDefaultOptions() {
        return helper.deepExtend(
            super.getDefaultOptions(),
            {
                margin: { top: 20, right: 20, bottom: 20, left: 20 },
                initialWidth: 800,
                initialHeight: 600
            }
        );
    }
    // ... snip: the rest of the example
}
```

My code editor (Sublime) complains "Cannot find module 'd3kit'" and this is backed up by the output when I try to compile (`$ tsc index.ts`) from the command line. The type definitions seem to be working fine as I can get code completion in my editor for d3kit classes and functions.

I've tried this on 2 computers with the same result. One using node 7.0.0, the other 6.7.0. Both have TypeScript v2 installed. I've verified that d3kit v3.1.1 was downloaded and is installed in `node_modules` in the place you'd expect it to be, so I'm not sure why the module loader would have trouble finding it. I've tried it both with and without importing the type definitions. I've also tried adding `/// <reference path="typings/index.d.ts" />` to the top of the file. No joy.

I confess the whole UMD paradigm has always flummoxed me so I can't tell if there is an issue with the way that the package is exported, but I'm frustrated that I can't seem to get a simple example to work.

Any ideas? I really like this package.

**Comment by kristw** [comment]
I never use TypeScript myself, so I probably know about the problem less than you. I suspect it might be something to do with the type description file. 

I tried remove @types/d3 from the dependencies and got the same error with d3

```bash
>> tsc index.ts --moduleResolution node
index.ts(3,21): error TS2307: Cannot find module 'd3'.
```

index.js

```js
import * as d3 from 'd3';
const x = d3.max([1,2]);
```

**Comment by morphatic** [comment]
You were right! It took a lot of fiddling, but I finally got the correct type definitions. 

Here is [the PR where I submitted the TypeScript definition updates](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/12713).

[Here is an example project that uses d3kit with TypeScript to build a re-usable chart](https://github.com/morphatic/d3kit_test).

In coming days/weeks, I expect to have some examples showing incorporation of d3kit into Angular2/Ionic2 projects using TypeScript.
